clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerServiceDelegator.java

```diff
@@ -83,39 +110,41 @@ public SharedKafkaConsumer assignConsumerFor(PubSubTopic versionTopic, PubSubTop
 
   @Override
   public void unsubscribeAll(PubSubTopic versionTopic) {
-    defaultConsumerService.unsubscribeAll(versionTopic);
-    if (consumerServiceForAAWCLeader != null) {
-      consumerServiceForAAWCLeader.unsubscribeAll(versionTopic);
-    }
+    consumerServices.forEach(kafkaConsumerService -> kafkaConsumerService.unsubscribeAll(versionTopic));
+    vtTopicPartitionPairToConsumerService.entrySet().removeIf(entry -> entry.getKey().getFirst().equals(versionTopic));
     storeVersionAAWCFlagMap.remove(versionTopic.getName());
   }
 
   @Override
   public void unSubscribe(PubSubTopic versionTopic, PubSubTopicPartition pubSubTopicPartition) {
-    getKafkaConsumerService(versionTopic, pubSubTopicPartition).unSubscribe(versionTopic, pubSubTopicPartition);
+    KafkaConsumerService kafkaConsumerService = getKafkaConsumerService(versionTopic, pubSubTopicPartition);
+    if (kafkaConsumerService != null) {
+      kafkaConsumerService.unSubscribe(versionTopic, pubSubTopicPartition);
+      vtTopicPartitionPairToConsumerService.remove(new Pair<>(versionTopic, pubSubTopicPartition));
+    }
```

Comment:
Shall we log a warn message for `else` condition?
It is unexpected, and this comment applies to all the related functions in this class.